lib/galaxy/webapps/galaxy/api/users.py

```diff
@@ -669,7 +669,7 @@ def set_toolbox_filters(self, trans, id, payload=None, **kwd):
         for filter_type in filter_types:
             new_filters = []
             for prefixed_name in payload:
-                if payload.get(prefixed_name) == 'true' and prefixed_name.startswith(filter_type):
+                if bool(payload.get(prefixed_name)) and prefixed_name.startswith(filter_type):
```

Comment:
If you want to do this I would suggest using string_as_bool, but why can't we pass in a boolean here and raise a RequestParameterInvalidException if it is not a boolean? Same for the cases below.

Reply:
We are passing a `boolean` here, it is not a string anymore but this still allows users to not pass a value at all e.g. `None` which will then default to `False`.

Reply:
If it's None you don't need to cast to a boolean, I think that'd be less confusing.

Reply:
Considering your comment, I am thinking that it might be best to use `util.string_as_bool` then it would remain backward compatible.

Reply:
please don't do that, we want to create a consistent API and we'll place parameter validation on the inputs.